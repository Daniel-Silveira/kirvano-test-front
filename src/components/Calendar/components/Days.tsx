import dayjs from "dayjs";
import { useCallback, useContext } from "react";

import DatepickerContext from "../../../contexts/DatepickerContext";
import {
  formatDate,
  nextMonth,
  previousMonth,
  classNames as cn,
} from "../../../helpers/calendar";
type DateType = "current" | "previous" | "next";

interface Props {
  calendarData: {
    date: dayjs.Dayjs;
    days: {
      previous: number[];
      current: number[];
      next: number[];
    };
  };
  onClickPreviousDays: (day: number) => void;
  onClickDay: (day: number) => void;
  onClickNextDays: (day: number) => void;
}

export const useDays = ({
  calendarData,
  onClickDay,
  onClickNextDays,
  onClickPreviousDays,
}: Props) => {
  const { period, changePeriod, dayHover, changeDayHover } =
    useContext(DatepickerContext);

  const currentDateClass = useCallback(
    (item: number): string => {
      const { date } = calendarData;
      const formattedItem = item >= 10 ? item : `0${item}`;
      const itemDate = `${date.year()}-${date.month() + 1}-${formattedItem}`;

      const currentDate = dayjs();
      const itemDateFormatted = dayjs(itemDate);

      if (formatDate(currentDate) !== formatDate(itemDateFormatted)) return "";

      return "text-primary";
    },
    [calendarData.date]
  );

  const activeDateData = useCallback(
    (day: number) => {
      const { date } = calendarData;
      const fullDay = `${date.year()}-${date.month() + 1}-${day}`;
      let className = "";
      const isStart = dayjs(fullDay).isSame(period.start);
      const isEnd = dayjs(fullDay).isSame(period.end);
      const isHover = dayjs(fullDay).isSame(dayHover);
      const isStartHover = isStart && isHover && !period.end;
      const isEndHover = isEnd && isHover && !period.start;

      if (isStart && isEnd) {
        className = ` bg-primary text-white font-medium rounded-lg`;
      } else if (isStart) {
        className = ` bg-primary text-white font-medium ${
          isStartHover ? "rounded-lg" : "rounded-l-lg"
        }`;
      } else if (isEnd) {
        className = ` bg-primary text-white font-medium ${
          isEndHover ? "rounded-lg" : "rounded-r-lg"
        }`;
      }

      return {
        active: isStart || isEnd,
        className: className,
      };
    },
    [calendarData.date, dayHover, period.end, period.start]
  );

  const hoverClassByDay = useCallback(
    (day: number) => {
      const fullDay = `${calendarData.date.year()}-${calendarData.date.month() + 1}-${day >= 10 ? day : "0" + day}`;
      let className = currentDateClass(day);

      if (period.start && period.end) {
        if (dayjs(fullDay).isBetween(period.start, period.end, "day", "[)")) {
          return ` bg-primary ${className} dark:bg-white/10`;
        }
      }

      if (!dayHover) {
        return className;
      }

      if (
        period.start &&
        dayjs(fullDay).isBetween(period.start, dayHover, "day", "[)")
      ) {
        className = ` bg-primary ${className} dark:bg-white/10`;
      }

      if (
        period.end &&
        dayjs(fullDay).isBetween(dayHover, period.end, "day", "[)")
      ) {
        className = ` bg-primary ${className} dark:bg-white/10`;
      }

      if (dayHover === fullDay) {
        className = ` transition-all duration-500 text-white font-medium bg-primary ${period.start ? "rounded-r-lg" : "rounded-l-lg"}`;
      }

      return className;
    },
    [calendarData.date, currentDateClass, dayHover, period.end, period.start]
  );

  const getMetaData = useCallback(() => {
    const { date } = calendarData;
    return {
      previous: previousMonth(date),
      current: date,
      next: nextMonth(date),
    };
  }, [calendarData.date]);

  const buttonClass = useCallback(
    (day: number, type: "current" | "next" | "previous") => {
      const baseClass =
        "flex items-center justify-center w-12 h-12 lg:w-10 lg:h-10";
      if (type === "current") {
        return cn(
          baseClass,
          !activeDateData(day).active
            ? hoverClassByDay(day)
            : activeDateData(day).className
        );
      }
      return cn(baseClass, "text-gray-400");
    },
    [activeDateData, hoverClassByDay]
  );

  const hoverDay = useCallback(
    (day: number, type: string) => {
      const object = getMetaData();
      const newDate = object[type as keyof typeof object];
      const newHover = `${newDate.year()}-${newDate.month() + 1}-${
        day >= 10 ? day : "0" + day
      }`;

      if (period.start && !period.end) {
        const hoverPeriod = { ...period, end: newHover };
        if (dayjs(newHover).isBefore(dayjs(period.start))) {
          hoverPeriod.start = newHover;
          hoverPeriod.end = period.start;
          changePeriod({
            start: null,
            end: period.start,
          });
        }
        changeDayHover(newHover);
      }

      if (!period.start && period.end) {
        const hoverPeriod = { ...period, start: newHover };
        if (dayjs(newHover).isAfter(dayjs(period.end))) {
          hoverPeriod.start = period.end;
          hoverPeriod.end = newHover;
          changePeriod({
            start: period.end,
            end: null,
          });
        }
        changeDayHover(newHover);
      }
    },
    [changeDayHover, changePeriod, getMetaData, period]
  );

  const handleClickDay = useCallback(
    (day: number, type: DateType) => {
      function continueClick() {
        const typesToFunction = {
          previous: () => onClickPreviousDays(day),
          current: () => onClickDay(day),
          next: () => onClickNextDays(day),
        };

        typesToFunction[type]();
      }

      continueClick();
    },
    [
      dayHover,
      getMetaData,
      onClickDay,
      onClickNextDays,
      onClickPreviousDays,
      period.end,
      period.start,
    ]
  );

  return {
    buttonClass,
    handleClickDay,
    hoverDay,
  };
};
