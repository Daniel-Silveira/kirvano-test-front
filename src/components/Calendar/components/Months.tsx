import dayjs from "dayjs";

import { LANGUAGE, MONTHS } from "../../../constants";
import { RoundedButton } from "./shared/RoundedButton";

interface Props {
  currentMonth: number;
  onClick: (month: number) => void;
}

const Months: React.FC<Props> = ({ currentMonth, onClick }) => {
  return (
    <div className="w-full grid grid-cols-2 gap-2 mt-2">
      {MONTHS.map((item) => (
        <RoundedButton
          key={item}
          padding="py-3"
          onClick={() => {
            onClick(item);
          }}
          active={currentMonth === item}
        >
          <>{dayjs(`2024-${item}-01`).locale(LANGUAGE).format("MMM")}</>
        </RoundedButton>
      ))}
    </div>
  );
};

export default Months;
