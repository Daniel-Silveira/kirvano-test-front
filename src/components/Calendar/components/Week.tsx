import dayjs from "dayjs";
import { useMemo } from "react";

import { DAYS, LANGUAGE } from "../../../constants";
import { shortString, ucFirst } from "../../../helpers/calendar";

type WeekDay = "mon" | "tue" | "wed" | "thu" | "fri" | "sat" | "sun";

const Week: React.FC = () => {
  const startWeekOn = "sun";
  const startDateModifier = useMemo(() => {
    const startWeekOnMap: Record<WeekDay, number> = {
      mon: 1,
      tue: 2,
      wed: 3,
      thu: 4,
      fri: 5,
      sat: 6,
      sun: 0,
    };

    return (startWeekOnMap as Record<string, number>)[startWeekOn] || 0;
  }, [startWeekOn]);

  return (
    <div className="grid grid-cols-7 py-2">
      {DAYS.map((item) => (
        <div
          key={item}
          className={`tracking-wide text-gray-500 text-center ${(item === 0 || item === 6) && "text-red-400"}`}
        >
          {ucFirst(
            shortString(
              dayjs(`2022-11-${6 + (item + startDateModifier)}`)
                .locale(LANGUAGE)
                .format("ddd")
            )
          )}
        </div>
      ))}
    </div>
  );
};

export default Week;
