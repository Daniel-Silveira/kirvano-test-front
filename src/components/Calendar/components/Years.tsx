import { generateArrayNumber } from "../../../helpers/calendar";
import { RoundedButton } from "./shared/RoundedButton";

interface Props {
  year: number;
  currentYear: number;
  minYear: number | null;
  maxYear: number | null;
  onClick: (data: number) => void;
}

const Years: React.FC<Props> = ({
  year,
  currentYear,
  minYear,
  maxYear,
  onClick,
}) => {
  const dateLooking = "forward";
  const setupDate = {
    backward: {
      startDate: year - 11,
      endDate: year,
    },
    middle: {
      startDate: year - 4,
      endDate: year + 7,
    },
    forward: {
      startDate: year,
      endDate: year + 11,
    },
  };

  const startDate = setupDate[dateLooking].startDate;
  const endDate = setupDate[dateLooking].endDate;

  return (
    <div className="w-full grid grid-cols-2 gap-2 mt-2">
      {generateArrayNumber(startDate, endDate).map((item, index) => (
        <RoundedButton
          key={index}
          padding="py-3"
          onClick={() => {
            onClick(item);
          }}
          active={currentYear === item}
          disabled={
            (maxYear !== null && item > maxYear) ||
            (minYear !== null && item < minYear)
          }
        >
          <>{item}</>
        </RoundedButton>
      ))}
    </div>
  );
};

export default Years;
