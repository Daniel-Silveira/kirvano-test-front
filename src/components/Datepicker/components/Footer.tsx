import { useContext } from "react";
import { DATE_FORMAT } from "../../../constants";
import { Button } from "../../shared/Button";

import dayjs from "dayjs";
import DatepickerContext from "../../../contexts/DatepickerContext";

const Footer: React.FC = () => {
  const { toggleVisibilityDatepicker, period, changeDatepickerValue } =
    useContext(DatepickerContext);

  return (
    <div className="flex items-center justify-end pb-2.5 pt-3">
      <div className="w-full md:w-auto flex items-center justify-center space-x-3">
        <Button
          onClick={() => {
            toggleVisibilityDatepicker();
          }}
          label="Cancel"
        />
        <Button
          onClick={() => {
            if (period.start && period.end) {
              changeDatepickerValue({
                startDate: dayjs(period.start).format(DATE_FORMAT),
                endDate: dayjs(period.end).format(DATE_FORMAT),
              });
              toggleVisibilityDatepicker();
            }
          }}
          label="Apply"
          primary
        />
      </div>
    </div>
  );
};

export default Footer;
