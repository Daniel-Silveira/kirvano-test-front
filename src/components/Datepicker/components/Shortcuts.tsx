import dayjs from "dayjs";
import { memo, useCallback, useContext } from "react";
import { Period, ShortcutsItem } from "../../../types";

import DEFAULT_SHORTCUTS from "../../../constants/shortcuts";
import DatepickerContext from "../../../contexts/DatepickerContext";

interface ItemTemplateProps {
  children: JSX.Element;
  key: number;
  item: { period: Period | any };
}

const ItemTemplate = memo((props: ItemTemplateProps) => {
  const {
    period,
    changePeriod,
    updateFirstDate,
    dayHover,
    changeDayHover,
    changeDatepickerValue,
  } = useContext(DatepickerContext);

  const chosePeriod = useCallback(
    (item: Period) => {
      if (dayHover) {
        changeDayHover(null);
      }
      if (period.start || period.end) {
        changePeriod({
          start: null,
          end: null,
        });
      }
      changePeriod(item);
      changeDatepickerValue &&
        changeDatepickerValue({
          startDate: item.start,
          endDate: item.end,
        });
      updateFirstDate(dayjs(item.start));
    },
    [
      changeDatepickerValue,
      changeDayHover,
      changePeriod,
      dayHover,
      period.end,
      period.start,
      updateFirstDate,
    ]
  );

  const children = props?.children;

  return (
    <li
      className="whitespace-nowrap w-1/2 md:w-1/3 lg:w-auto transition-all duration-300 hover:bg-gray-100 dark:hover:bg-white/10 p-2 rounded cursor-pointer text-zinc-400"
      onClick={() => {
        chosePeriod(props?.item.period);
      }}
    >
      {children}
    </li>
  );
});

const Shortcuts: React.FC = () => {
  const shortcutOptions = Object.entries(DEFAULT_SHORTCUTS);

  const printItemText = useCallback((item: ShortcutsItem) => {
    return item?.text ?? null;
  }, []);

  return shortcutOptions?.length ? (
    <div className="mb-3 lg:mb-0  pr-1">
      <ul className="w-full tracking-wide flex flex-wrap lg:flex-col p-3">
        {shortcutOptions.map(([, item], index: number) =>
          Array.isArray(item) ? (
            item.map((item, index) => (
              <ItemTemplate key={index} item={item}>
                <>{item.text}</>
              </ItemTemplate>
            ))
          ) : (
            <ItemTemplate key={index} item={item}>
              <>{printItemText(item)}</>
            </ItemTemplate>
          )
        )}
      </ul>
    </div>
  ) : null;
};

export default Shortcuts;
