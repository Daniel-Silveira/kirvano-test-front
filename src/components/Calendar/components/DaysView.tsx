import dayjs from "dayjs";
import isBetween from "dayjs/plugin/isBetween";
import { useDays } from "./Days";

dayjs.extend(isBetween);

interface Props {
  calendarData: {
    date: dayjs.Dayjs;
    days: {
      previous: number[];
      current: number[];
      next: number[];
    };
  };
  onClickPreviousDays: (day: number) => void;
  onClickDay: (day: number) => void;
  onClickNextDays: (day: number) => void;
}

const Days: React.FC<Props> = (props) => {
  const { previous, current, next } = props.calendarData.days;

  return (
    <div className="grid grid-cols-7 gap-y-0.5 my-1">
      {previous.map((item, index) => (
        <DateButton key={index} day={item} type="previous" {...props} />
      ))}

      {current.map((item, index) => (
        <DateButton key={index} day={item} type="current" {...props} />
      ))}

      {next.map((item, index) => (
        <DateButton key={index} day={item} type="next" {...props} />
      ))}
    </div>
  );
};

interface DateButtonProps extends Props {
  day: number;
  type: "previous" | "current" | "next";
}

const DateButton: React.FC<DateButtonProps> = ({
  day,
  type,
  calendarData,
  onClickPreviousDays,
  onClickDay,
  onClickNextDays,
}) => {
  const { buttonClass, handleClickDay, hoverDay } = useDays({
    calendarData,
    onClickPreviousDays,
    onClickDay,
    onClickNextDays,
  });

  return (
    <button
      type="button"
      className={buttonClass(day, type)}
      onClick={() => handleClickDay(day, type)}
      onMouseOver={() => {
        hoverDay(day, type);
      }}
    >
      {day}
    </button>
  );
};

export default Days;
