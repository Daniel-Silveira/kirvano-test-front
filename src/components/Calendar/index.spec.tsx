import { render, fireEvent, renderHook } from "@testing-library/react";
import Years from "./components/Years";
import Week from "./components/Week";
import Months from "./components/Months";
import { useDays } from "./components/Days";
import dayjs from "dayjs";
import Days from "./components/DaysView";

describe("[Calendar] components", () => {
  describe("Years component", () => {
    const mockOnClick = jest.fn();

    it("should render years within the specified range", () => {
      const { container } = render(
        <Years
          year={2010}
          currentYear={2024}
          minYear={2010}
          maxYear={2030}
          onClick={mockOnClick}
        />
      );

      expect(container).toMatchSnapshot();
    });

    it("should call onClick when a year is clicked", () => {
      const { getByText } = render(
        <Years
          year={2024}
          currentYear={2024}
          minYear={2000}
          maxYear={2030}
          onClick={mockOnClick}
        />
      );

      const yearElement = getByText("2024");
      fireEvent.click(yearElement);

      expect(mockOnClick).toHaveBeenCalledWith(2024);
    });

    it("should disable years outside the minYear and maxYear range", () => {
      const { getByText, queryByText } = render(
        <Years
          year={2024}
          currentYear={2024}
          minYear={2025}
          maxYear={2030}
          onClick={mockOnClick}
        />
      );

      const disabledYear = getByText("2024");
      const enabledYear = queryByText("2025");

      expect(disabledYear).toBeDisabled();
      expect(enabledYear).not.toBeDisabled();
    });
  });

  describe("Week component", () => {
    it('should render days of the week starting from Sunday when startWeekOn is set to "sun"', () => {
      const { getByText } = render(<Week />);

      const days = ["Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"];
      days.forEach((day) => {
        const dayElement = getByText(day);
        expect(dayElement).toBeInTheDocument();
      });
    });
  });

  describe("Months component", () => {
    const mockOnClick = jest.fn();

    it("should render buttons for all months", () => {
      const { container } = render(
        <Months currentMonth={1} onClick={mockOnClick} />
      );

      expect(container).toMatchSnapshot();
    });

    it("should call onClick with the correct month when a month button is clicked", () => {
      const { getByText } = render(
        <Months currentMonth={1} onClick={mockOnClick} />
      );

      const februaryButton = getByText("Feb");
      fireEvent.click(februaryButton);

      expect(mockOnClick).toHaveBeenCalledWith(2);
    });
  });

  describe("useDays hook", () => {
    const calendarData: any = {
      date: dayjs("2022-11-01"),
      days: {
        previous: [],
        current: [1, 2, 3, 4, 5, 6, 7],
        next: [],
      },
    };

    const onClickPreviousDays = jest.fn();
    const onClickDay = jest.fn();
    const onClickNextDays = jest.fn();

    it("should return buttonClass, handleClickDay, and hoverDay functions", () => {
      const { result } = renderHook(() =>
        useDays({
          calendarData,
          onClickPreviousDays,
          onClickDay,
          onClickNextDays,
        })
      );

      expect(result.current.buttonClass).toBeDefined();
      expect(result.current.handleClickDay).toBeDefined();
      expect(result.current.hoverDay).toBeDefined();
    });
  });
  describe("Days component", () => {
    const mockCalendarData = {
      date: dayjs("2022-11-01"),
      days: {
        previous: [27, 28, 29, 30],
        current: [1, 2, 3, 4, 5, 6, 7],
        next: [8, 9, 10, 11, 12],
      },
    };

    const mockOnClickPreviousDays = jest.fn();
    const mockOnClickDay = jest.fn();
    const mockOnClickNextDays = jest.fn();

    it("should render buttons for all days", () => {
      const { getByText } = render(
        <Days
          calendarData={mockCalendarData}
          onClickPreviousDays={mockOnClickPreviousDays}
          onClickDay={mockOnClickDay}
          onClickNextDays={mockOnClickNextDays}
        />
      );

      mockCalendarData.days.previous.forEach((day) => {
        expect(getByText(day.toString())).toBeInTheDocument();
      });

      mockCalendarData.days.current.forEach((day) => {
        expect(getByText(day.toString())).toBeInTheDocument();
      });

      mockCalendarData.days.next.forEach((day) => {
        expect(getByText(day.toString())).toBeInTheDocument();
      });
    });

    it("should call onClickDay when a day button is clicked", () => {
      const { getByText } = render(
        <Days
          calendarData={mockCalendarData}
          onClickPreviousDays={mockOnClickPreviousDays}
          onClickDay={mockOnClickDay}
          onClickNextDays={mockOnClickNextDays}
        />
      );

      fireEvent.click(getByText("1"));

      expect(mockOnClickDay).toHaveBeenCalledWith(1);
    });

    it("should call hoverDay when mouse is over a day button", () => {
      const { getByText } = render(
        <Days
          calendarData={mockCalendarData}
          onClickPreviousDays={mockOnClickPreviousDays}
          onClickDay={mockOnClickDay}
          onClickNextDays={mockOnClickNextDays}
        />
      );

      fireEvent.mouseOver(getByText("1"));
    });
  });
});
