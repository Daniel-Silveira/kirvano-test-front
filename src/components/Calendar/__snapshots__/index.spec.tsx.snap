// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Calendar] components Months component should render buttons for all months 1`] = `
<div>
  <div
    class="w-full grid grid-cols-2 gap-2 mt-2"
  >
    <button
      class="w-full tracking-wide dark:text-white/70 dark:hover:bg-white/10 dark:focus:bg-white/10 font-semibold bg-gray-50 dark:bg-white/5 transition-all duration-300 px-3 py-3 uppercase hover:bg-gray-100 rounded-md focus:ring-1 "
      type="button"
    >
      Jan
    </button>
    <button
      class="w-full tracking-wide dark:text-white/70 dark:hover:bg-white/10 dark:focus:bg-white/10  transition-all duration-300 px-3 py-3 uppercase hover:bg-gray-100 rounded-md focus:ring-1 "
      type="button"
    >
      Feb
    </button>
    <button
      class="w-full tracking-wide dark:text-white/70 dark:hover:bg-white/10 dark:focus:bg-white/10  transition-all duration-300 px-3 py-3 uppercase hover:bg-gray-100 rounded-md focus:ring-1 "
      type="button"
    >
      Mar
    </button>
    <button
      class="w-full tracking-wide dark:text-white/70 dark:hover:bg-white/10 dark:focus:bg-white/10  transition-all duration-300 px-3 py-3 uppercase hover:bg-gray-100 rounded-md focus:ring-1 "
      type="button"
    >
      Apr
    </button>
    <button
      class="w-full tracking-wide dark:text-white/70 dark:hover:bg-white/10 dark:focus:bg-white/10  transition-all duration-300 px-3 py-3 uppercase hover:bg-gray-100 rounded-md focus:ring-1 "
      type="button"
    >
      May
    </button>
    <button
      class="w-full tracking-wide dark:text-white/70 dark:hover:bg-white/10 dark:focus:bg-white/10  transition-all duration-300 px-3 py-3 uppercase hover:bg-gray-100 rounded-md focus:ring-1 "
      type="button"
    >
      Jun
    </button>
    <button
      class="w-full tracking-wide dark:text-white/70 dark:hover:bg-white/10 dark:focus:bg-white/10  transition-all duration-300 px-3 py-3 uppercase hover:bg-gray-100 rounded-md focus:ring-1 "
      type="button"
    >
      Jul
    </button>
    <button
      class="w-full tracking-wide dark:text-white/70 dark:hover:bg-white/10 dark:focus:bg-white/10  transition-all duration-300 px-3 py-3 uppercase hover:bg-gray-100 rounded-md focus:ring-1 "
      type="button"
    >
      Aug
    </button>
    <button
      class="w-full tracking-wide dark:text-white/70 dark:hover:bg-white/10 dark:focus:bg-white/10  transition-all duration-300 px-3 py-3 uppercase hover:bg-gray-100 rounded-md focus:ring-1 "
      type="button"
    >
      Sep
    </button>
    <button
      class="w-full tracking-wide dark:text-white/70 dark:hover:bg-white/10 dark:focus:bg-white/10  transition-all duration-300 px-3 py-3 uppercase hover:bg-gray-100 rounded-md focus:ring-1 "
      type="button"
    >
      Oct
    </button>
    <button
      class="w-full tracking-wide dark:text-white/70 dark:hover:bg-white/10 dark:focus:bg-white/10  transition-all duration-300 px-3 py-3 uppercase hover:bg-gray-100 rounded-md focus:ring-1 "
      type="button"
    >
      Nov
    </button>
    <button
      class="w-full tracking-wide dark:text-white/70 dark:hover:bg-white/10 dark:focus:bg-white/10  transition-all duration-300 px-3 py-3 uppercase hover:bg-gray-100 rounded-md focus:ring-1 "
      type="button"
    >
      Dec
    </button>
  </div>
</div>
`;

exports[`[Calendar] components Years component should render years within the specified range 1`] = `
<div>
  <div
    class="w-full grid grid-cols-2 gap-2 mt-2"
  >
    <button
      class="w-full tracking-wide dark:text-white/70 dark:hover:bg-white/10 dark:focus:bg-white/10  transition-all duration-300 px-3 py-3 uppercase hover:bg-gray-100 rounded-md focus:ring-1 "
      type="button"
    >
      2010
    </button>
    <button
      class="w-full tracking-wide dark:text-white/70 dark:hover:bg-white/10 dark:focus:bg-white/10  transition-all duration-300 px-3 py-3 uppercase hover:bg-gray-100 rounded-md focus:ring-1 "
      type="button"
    >
      2011
    </button>
    <button
      class="w-full tracking-wide dark:text-white/70 dark:hover:bg-white/10 dark:focus:bg-white/10  transition-all duration-300 px-3 py-3 uppercase hover:bg-gray-100 rounded-md focus:ring-1 "
      type="button"
    >
      2012
    </button>
    <button
      class="w-full tracking-wide dark:text-white/70 dark:hover:bg-white/10 dark:focus:bg-white/10  transition-all duration-300 px-3 py-3 uppercase hover:bg-gray-100 rounded-md focus:ring-1 "
      type="button"
    >
      2013
    </button>
    <button
      class="w-full tracking-wide dark:text-white/70 dark:hover:bg-white/10 dark:focus:bg-white/10  transition-all duration-300 px-3 py-3 uppercase hover:bg-gray-100 rounded-md focus:ring-1 "
      type="button"
    >
      2014
    </button>
    <button
      class="w-full tracking-wide dark:text-white/70 dark:hover:bg-white/10 dark:focus:bg-white/10  transition-all duration-300 px-3 py-3 uppercase hover:bg-gray-100 rounded-md focus:ring-1 "
      type="button"
    >
      2015
    </button>
    <button
      class="w-full tracking-wide dark:text-white/70 dark:hover:bg-white/10 dark:focus:bg-white/10  transition-all duration-300 px-3 py-3 uppercase hover:bg-gray-100 rounded-md focus:ring-1 "
      type="button"
    >
      2016
    </button>
    <button
      class="w-full tracking-wide dark:text-white/70 dark:hover:bg-white/10 dark:focus:bg-white/10  transition-all duration-300 px-3 py-3 uppercase hover:bg-gray-100 rounded-md focus:ring-1 "
      type="button"
    >
      2017
    </button>
    <button
      class="w-full tracking-wide dark:text-white/70 dark:hover:bg-white/10 dark:focus:bg-white/10  transition-all duration-300 px-3 py-3 uppercase hover:bg-gray-100 rounded-md focus:ring-1 "
      type="button"
    >
      2018
    </button>
    <button
      class="w-full tracking-wide dark:text-white/70 dark:hover:bg-white/10 dark:focus:bg-white/10  transition-all duration-300 px-3 py-3 uppercase hover:bg-gray-100 rounded-md focus:ring-1 "
      type="button"
    >
      2019
    </button>
    <button
      class="w-full tracking-wide dark:text-white/70 dark:hover:bg-white/10 dark:focus:bg-white/10  transition-all duration-300 px-3 py-3 uppercase hover:bg-gray-100 rounded-md focus:ring-1 "
      type="button"
    >
      2020
    </button>
    <button
      class="w-full tracking-wide dark:text-white/70 dark:hover:bg-white/10 dark:focus:bg-white/10  transition-all duration-300 px-3 py-3 uppercase hover:bg-gray-100 rounded-md focus:ring-1 "
      type="button"
    >
      2021
    </button>
  </div>
</div>
`;
